File: bot/empty_prompt.py
import re

from bot.commands import all_commands

def is_empty_prompt(prompt: str) -> bool:
    stripped_prompt = prompt.strip()
    return (stripped_prompt in all_commands) or \
        (re.match(r'^/[a-zA-Z]+$', stripped_prompt)) or \
        (re.match(r'^1:(midjourney|dalle|flux|sd|suno):.+$', stripped_prompt)) or \
        (not any(char.isalnum() for char in stripped_prompt))

File: js/bot/empty_prompt.js
import { ALL_COMMANDS } from './commands.js';

// Checks if prompt is empty or just a command
export function isEmptyPrompt(prompt) {
  const stripped = prompt.trim();
  // Check if prompt is exactly a bot command
  if (ALL_COMMANDS.includes(stripped)) return true;
  // Check single slash commands like /start
  if (/^\/[a-zA-Z]+$/.test(stripped)) return true;
  // Check internal commands of form 1:midjourney:...
  if (/^1:(midjourney|dalle|flux|sd|suno):.+$/.test(stripped)) return true;
  // Check if no alphanumeric characters
  if (!/[A-Za-z0-9]/.test(stripped)) return true;
  return false;
}
